packages/fx-core/src/plugins/solution/fx-solution/utils/depsChecker/bicepChecker.ts

```diff
@@ -0,0 +1,356 @@
+      await this.closeStream(bicepWriter);
+    }
+  }
+
+  private async writeBicepBits(writer: Writable, reader: Readable): Promise<void> {
+    return new Promise((resolve: (value: void) => void, reject: (e: Error) => void): void => {
+      reader.pipe(writer);
+      finished(writer, (err?: NodeJS.ErrnoException | null) => {
+        if (err) reject(err);
+        else resolve();
+      });
+    });
+  }
+
+  private async closeStream(writer: Writable): Promise<void> {
+    return new Promise((resolve: (value: void) => void): void => {
+      writer.end(() => resolve());
+    });
+  }
+
+  private async validate(): Promise<boolean> {
+    let isVersionSupported = false;
+    let privateVersion = "";
+    try {
+      privateVersion = await this.queryVersion(this.getBicepExecPath());
+      isVersionSupported = this.isVersionSupported(privateVersion);
+    } catch (err) {
+      sendSystemErrorEvent(
+        DepsCheckerEvent.bicepValidationError,
+        TelemtryMessages.failedToValidateBicep,
+        err,
+        this._telemetry
+      );
+      await this._logger?.error(`${TelemtryMessages.failedToValidateBicep}, error = ${err}`);
+    }
+
+    if (!isVersionSupported) {
+      this._telemetry?.sendTelemetryEvent(DepsCheckerEvent.bicepValidationError, {
+        ...{ "bicep-private-version": privateVersion },
+        ...getCommonProps(),
+      });
+    }
+
+    return isVersionSupported;
+  }
+
+  private async handleInstallCompleted() {
+    this._telemetry?.sendTelemetryEvent(DepsCheckerEvent.bicepInstallCompleted);
+    await this._logger?.info(Messages.finishInstallBicep.replace("@NameVersion", displayBicepName));
+  }
+
+  private async handleInstallFailed(): Promise<void> {
+    await this.cleanup();
+    this._telemetry?.sendTelemetryEvent(DepsCheckerEvent.bicepInstallError);
+    throw new SystemError(
+      DepsCheckerEvent.bicepInstallError,
+      Messages.failToInstallBicep.split("@NameVersion").join(displayBicepName),
+      source
+    );
+  }
+
+  private isVersionSupported(version: string): boolean {
+    return supportedVersions.some((supported) => version.includes(supported));
+  }
+
+  private async isBicepInstalled(path: string): Promise<boolean> {
+    try {
+      const version = await this.queryVersion(path);
+      return this.isVersionSupported(version);
+    } catch (e) {
+      // do nothing
+      return false;
+    }
+  }
+
+  public async getBicepCommand(): Promise<string> {
+    if (await this.isInstalled()) {
+      return this.getBicepExecPath();
+    }
+    return "bicep";
+  }
+
+  private getBicepExecPath(): string {
+    return path.join(this.getBicepInstallDir(), this.getBicepFileName());
+  }
+
+  private getBicepFileName(): string {
+    if (isWindows()) {
+      return "bicep.exe";
+    }
+    return "bicep";
+  }
+
+  private getBicepBitSuffixName(): string {
+    if (isWindows()) {
+      return "bicep-win-x64.exe";
+    }
+    if (isMacOS()) {
+      return "bicep-osx-x64";
```

Comment:
Does bicep support M1? If not, maybe detect M1 Mac and explicitly tell it to the user

Reply:
> what is the empty file named `placeholder` for?

removed

Reply:
> Does bicep support M1? If not, maybe detect M1 Mac and explicitly tell it to the user

`bicep` work well on my mac m1, but it available for all mac m1 is to verify.
If bicep not support mac m1, the version of ARM need to have a workaround for m1 user.